# Задачи за домашно

Работете върху копие на дадения файл, като попълните с код клетките, маркирани с коментара ```# your code here```, така че получения от вас резлутат (output) да съвпада с дадения под съответната клетка с код.

## Задача 1

Използвайте данните в CSV файла даден тук: https://github.com/geekcourses/JupyterNotebooksExamples/blob/master/datasets/various/drinks.csv, като го свалите локално на вашата машина и заредите данните в DataFrame обект с име `df`.

Покажете първите 10 реда от `df`.

In [1]:
import pandas as pd

In [4]:
# your code here

# Не ми работи връзката т.е. правя го на слъжебния ком и нямам достъп тябва да включвам проксито което смятам е изклишно 
df = pd.read_csv('drinks.csv')
df.head(10)

country  beer_servings  spirit_servings  wine_servings  \
0        Afghanistan              0                0              0   
1            Albania             89              132             54   
2            Algeria             25                0             14   
3            Andorra            245              138            312   
4             Angola            217               57             45   
5  Antigua & Barbuda            102              128             45   
6          Argentina            193               25            221   
7            Armenia             21              179             11   
8          Australia            261               72            212   
9            Austria            279               75            191   

   total_litres_of_pure_alcohol continent  
0                           0.0        AS  
1                           4.9        EU  
2                           0.7        AF  
3                          12.4        EU  
4                           5.9        AF  
5                           4.9       NaN  
6                           8.3        SA  
7                           3.8        EU  
8                          10.4        OC  
9                           9.7        EU

## Задача 2

Покажете броя на редовете и колоните в `df`.

In [5]:
# your code here
df.shape

(193, 6)

## Задача 3

Покажете колко е броя на държавите в континент 'EU'

In [10]:
# your code here
df[df['continent'] == 'EU'].country.count()

np.int64(45)

## Задача 4

Покажете количествата бира ('beer_servings'), вино ('wine_servings') и спиртни напитки ('spirit_servings'), които се конcумират в Bulgaria

In [15]:
# your code here

df.iloc[25, 1:4]


beer_servings      231
spirit_servings    252
wine_servings       94
Name: 25, dtype: object

## Задача 5

Покажете какво е максимално количество консумирано вино ('wine_servings') в целия дейтасет и коя е държавата в която се консумира то.

In [25]:
# your code here

df[df.wine_servings ==  df['wine_servings'].max()]

country  beer_servings  spirit_servings  wine_servings  \
61  France            127              151            370   

    total_litres_of_pure_alcohol continent  
61                          11.8        EU

Mаксималното количество консумирано вино в целия дейтасет е 370 литра, консумирани във France.

## Задача 6

Покажете какво е максимално количество консумирано вино ('wine_servings') в континент Южна Америка ('SA') и коя е държавата в която се консумира то.

In [44]:
# your code here
df.loc[df[df.continent == 'SA']['wine_servings'].idxmax(), :]

country                         Argentina
beer_servings                         193
spirit_servings                        25
wine_servings                         221
total_litres_of_pure_alcohol          8.3
continent                              SA
Name: 6, dtype: object

Mаксималното количество консумирано вино в Южна Америка е 221 литра, консумирани във Argentina

## Задача 7

Покажете държавата в която се консумира най-много бира ('beer_servings'), какво е количеството и в кой континент се намира тя.

In [48]:
# your code here

df.iloc[df.beer_servings.idxmax(),:]

country                         Namibia
beer_servings                       376
spirit_servings                       3
wine_servings                         1
total_litres_of_pure_alcohol        6.8
continent                            AF
Name: 117, dtype: object

Държавата в която се консумира най-много бира  е Namibia - 376 литра. Намира се в AF.

## Задача 8

Покажете държавите в които не се консумира бира.

In [62]:
# your code here
df.country[df['beer_servings'] == 0]

0           Afghanistan
13           Bangladesh
40         Cook Islands
46          North Korea
79                 Iran
90               Kuwait
97                Libya
103            Maldives
106    Marshall Islands
107          Mauritania
111              Monaco
128            Pakistan
147          San Marino
149        Saudi Arabia
158             Somalia
Name: country, dtype: object